src/wallet/connection.ts

```diff
@@ -88,4 +91,85 @@ class WalletServiceConnection extends BaseConnection {
   }
 }
 
-export default WalletServiceConnection;
+/**
+ * This connection is used when the wallet has websocket disabled.
+ */
+export class DummyWalletServiceConnection implements IConnection {
+  public readonly isDummyConnection = true;
+  public walletId?: string;
+
+  private currentNetwork: string;
+  private currentServer: string;
+  protected state: ConnectionState;
+
+  constructor(options: WalletServiceConnectionParams = { walletId: 'dummy-wallet' }) {
+    const {
+      walletId,
+      network,
+      servers,
+    } = {
+      ...DEFAULT_PARAMS,
+      ...options,
+    };
+
+    if (!network) {
+      throw Error('You must explicitly provide the network.');
+    }
+
+    this.walletId = walletId;
+    this.currentNetwork = network || 'dummynet';
+    this.state = ConnectionState.CLOSED;
+    this.currentServer = servers[0] || config.getServerUrl();
+  }
+
+  /**
+   * Dummy method to simulate the EventEmitter.on method to avoid errors
+   * @param _event dummy event
+   * @param _listener dummy listener
+   * @returns dummy emitter
+   */
+  on(_event: string, _listener: (...args: any[]) => void): EventEmitter {
+    // return a dummy emitter that does nothing
+    return new EventEmitter();
+  }
+
+  start(): void {
+    // There is nothing to start
+  }
+
+  stop(): void {
+    // There is nothing to stop
+  }
+
+  endConnection(): void {
+    // There is no connection to end
+  }
+
+  setup(): void {
+    // There is nothing to setup
+  }
+
+  handleWalletMessage(_wsData: any): void {
+    // There is nothing to handle
+  }
+
+  onConnectionChange(value: boolean): void {
+    // There is nothing to change
+  }
+
+  setState(state: ConnectionState): void {
+    this.state = state;
+  }
+
+  getCurrentServer(): string {
+    return this.currentServer;
+  }
+
+  getCurrentNetwork(): string {
+    return this.currentNetwork;
+  }
+
+  setWalletId(walletId: string) {
+    this.walletId = walletId;
+  }
```

Comment:
I still don't understand why we need to implement those methods here, they are duplicated code from other classes.